packages/jsii-java-runtime-test/project/src/test/java/software/amazon/jsii/testing/ComplianceTest.java

```diff
@@ -1268,6 +1274,27 @@ public void canLoadEnumValues() {
         assertNotNull(EnumDispenser.randomIntegerLikeEnum());
     }
 
+    public void listInStructCorrectlyBuilds() {
+        StructWithCollections structWithCollections = StructWithCollections.builder()
+            .array(Arrays.asList("one", "two"))
+            .addToArray("three")
```

Comment:
What happens if a user has a property called `addToArray` in the same struct?

Reply:
Needs additional validation checks in jsii.

Reply:
To answer Elad's question literally, they will get a compile error in the generated Java code. I'll look into adding a validation check in jsii as recommended.

Reply:
What will the validation be? Not allow people to name a property `addToXxx`?

Reply:
... or putInXXX. We considered mangling the java implementation if a conflict was detected, but having both methods side by side on the builder could be extremely confusing. Ideally we'd detect in JSII if the collision was actually going to happen by doing an exact property name match but it's going to be cheaper to do a global 'any property that starts with addTo or putIn' check.

Reply:
Okay I've added in the JSII validation code. I also found and fixed an error with the pattern matching logic for error messages used in all the neg tests.

Reply:
Feels messy that we are leaking the builder pattern naming conventions to all Jsii code. I can totally imagine someone wanting to define a struct that has a Boolean property “addToFoo”... feels wrong. Why did we get rid of the withXxx prefix. It will provide a more robust way to handle things like this and also forward looking as it will allow adding additional verbs in the future.

Reply:
> Why did we get rid of the withXxx prefix

We were trying to emulate the style patterns currently used in the AWS Java v2 SDK and the popular libraries such as 'Immutables' that are strongly influencing the style of the larger Java community.

Alternatives we considered:
1) Allow conflicts in the model and then mangle the auto-generated method to not conflict on the java side. Eg: addToFoo(...) becomes addToFooArray(...) or something. The thing it falls back to could still theoretically conflict but we felt this was better than using a completely non-idiomatic form such as an underscore. We rejected this solution primarily on the grounds that seeing both methods side by side to each other in the builder may be very confusing.

2) Tune the conflict detection in JSII to only error in the specific case that an array is shadowing another property with the same name prefixed with addTo.

Having discussed this with the team we favor option 2 if the status-quo is unacceptable.

Reply:
> 1. Allow conflicts in the model and then mangle the auto-generated method to not conflict on the java side.

That unfortunately won't work because of library evolution.

Version 1 of the struct `{ foo: string[] }`, you'll generate an `addToFoo()` method in Java and people will use it.

Verstion 2 of the struct `{ foo: string[]; addToFoo: boolean; }`, the meaning of `addToFoo()` will have changed and the user's code will no longer compile.

It will have to be solution 2.

Reply:
Yeah that's a good point. We'd have to mangle the 'new' thing to avoid backwards-compatibility issues but it wouldn't ever be clear what the new thing was, as it could either be the base attribute or the addTo attribute. Either way I think we're all in agreement that wasn't going to be the path forward.